src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp

```diff
@@ -523,8 +518,8 @@ BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(
 
     Kokkos::Profiling::popRegion();
   }
-  // do not copy if by some miracle each query exactly yielded as many results
-  // as the buffer size
+  // Do not copy if each query exactly yielded as many results
+  // as the buffer size, or if both are zero.
```

Comment:
I would just omit it since you don't consider it in the if switch anyway. Alternatively, you could just do
```
if (n_results > 0)
{
  if (max_results_per_query > buffer_size)
  {
    ...
  }
  else if (n_results != static_cast<int>(n_queries) * buffer_size)
  {
    ...
  }
}
```
for clarity.

Reply:
I thought about doing that, but thought it may look good. Don't really care.